### PR TITLE
use randSecure and randABytes

### PR DESCRIPTION
**Pull request by BaoshanPang**
r? @n-salim

### CODE REVIEW

File: src/rngs/os.rs [hunk 1]
```diff
@@ -925,45 +925,46 @@ mod imp {
...
-
     impl OsRngImpl for OsRng {
         fn new() -> Result<OsRng, Error> { Ok(OsRng) }
 
         fn fill_chunk(&mut self, dest: &mut [u8]) -> Result<(), Error> {
+            static RNG_INIT: AtomicBool = AtomicBool::new(false);
+            while !RNG_INIT.load(Relaxed) {
+                let ret = unsafe { libc::randSecure() };
+                if ret < 0 {
+                    panic!("couldn't generate random bytes: {}", io::Error::last_os_error());
```
Comment (n-salim) [inline src/rngs/os.rs]: Should this really be a panic? Can't we return an error instead, as at line 955? E.g. see what is done here:

https://github.com/rust-random/getrandom/pull/86/files

**Review by n-salim — COMMENTED** [review]
Hello

File: src/rngs/os.rs (hunk 1 above)
Comment (BaoshanPang) [reply src/rngs/os.rs]: Code is updated.

File: src/rngs/os.rs [hunk 2]
```diff
@@ -925,30 +925,36 @@ mod imp {
...
-
     impl OsRngImpl for OsRng {
         fn new() -> Result<OsRng, Error> { Ok(OsRng) }
 
         fn fill_chunk(&mut self, dest: &mut [u8]) -> Result<(), Error> {
+            static RNG_INIT: AtomicBool = AtomicBool::new(false);
+            while !RNG_INIT.load(Relaxed) {
+                let ret = unsafe { libc::randSecure() };
+                if ret < 0 {
+                    Err(Error::with_cause(
```
Comment (n-salim) [inline src/rngs/os.rs]: Should be:

return Err(...)

right? Otherwise we'll just produce an expression and carry on looping.

Comment (BaoshanPang) [reply src/rngs/os.rs]: Oh, my bad.

**Review by n-salim — APPROVED** [review]

**Pull request merged**